src/tox_ansible/tox_molecule_case.py

```diff
@@ -67,12 +69,14 @@ def get_dependencies(self):
             dependencies.append("ansible=={}.*".format(self.ansible))
         else:
             dependencies.append("ansible")
-        # Drivers can have their own dependencies
-        if (
-            self.scenario.driver is not None
-            and self.scenario.driver in DRIVER_DEPENDENCIES.keys()
-        ):
-            dependencies.extend(DRIVER_DEPENDENCIES[self.scenario.driver])
+        # Drivers can have their own dependencies but we preinstall of known
+        # ones as the current venv may be shared among multiple scenarios
+        # If this proves to become heavy, we should look into finding a way
+        # to optimize it by allowing user to enable a slim mode.
+        for _, deps in DRIVER_DEPENDENCIES.items():
+            for dep in deps:
+                if dep not in dependencies:
+                    dependencies.append(dep)
```

Comment:
If this is flattening all of the dependencies into one, we can remove the redundancy between podman and docker that I put into the original deps list up at lines 12-13.

Reply:
Yeah. I know but i will postpone it for now. I have the impression that we will be revisiting the logic soon anyway.

For the moment I want to focus on bringing this plugin to production for tripleo repos. Once users will see how easy is to use, I expect to start seeing features requests being raised.